### DISCUSSION TITLE
Non accedo più a HA dal telefono.

### DISCUSSION THREAD

**Discussion by fraygon** [indomus/forum#553] [HUB: Home Assistant]
Da un po di giorni ho questo problema. In pratica se apro il pc collegato via LAN riesco a vedere HA sia tramite duckDNS che tramite ip locale ma se tento di aprirlo tramite l'APP dal telefono HA pubblica la notifica

Login attempt or request with invalid authentication from myfastgate.lan (192.168.x.xxx). See the log for details

e se leggo il log viene scritto questo:

2021-07-24 16:28:25 WARNING (MainThread) [homeassistant.components.http.ban] Login attempt or request with invalid authentication from myfastgate.lan (192.168.x.xxx). (Mozilla/5.0 (Linux; Android 7.0; HUAWEI VNS-L31 Build/HUAWEIVNS-L31; wv) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/91.0.4472.164 Mobile Safari/537.36)

Ho tentato via internet se vi è una soluzione ma senza esito. Da quando sono iniziati i problemi ho solo aggiornato core e esphome.  
Vi chiedo disperatamente aiuto. Dimenticavo che H.A. funziona in HASSIO.

**Comment by Scialla** [comment]
Usi un reverse proxy?

**Reply by fraygon** [reply]
ti chiedo pazienza ma in verità non so se ho un proxi inverso ma, da quello che ho capito penso di no. Non ci capisco molto di reti. Posso dirti che HA in Hassio gira da più di un anno. L'installazione l'ho fatta con la guida indomus quindi ho installato duckdns, ho fatto i certificati let's encript
<http:
  #base_url: https://mio xxx.duckdns.org:8123
  ssl_certificate: /ssl/fullchain.pem
  ssl_key: /ssl/privkey.pem
>
e tutto funzionava alla grande. Non riesco a capire se il modem fastweb è stato aggiornato di recente perchè non vedo date ma solo versione del firmaware.
Una cosa strana, che mi sono accorto ieri, è questa: riesco ad aprire HA dal telefono fuori casa tramite browser ma se tento tramite app mi dà il log descitto in precedenza. Ho provato a ricaricare la app senza esito. Anche la app installata su un tablet dà lo stesso risultato.
Oggi mi sono usciti altri log:
Login attempt or request with invalid authentication from 37.163.131.176 (37.163.131.176). (Mozilla/5.0 (Linux; Android 7.0; HUAWEI VNS-L31 Build/HUAWEIVNS-L31; wv) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/91.0.4472.164 Mobile Safari/537.36)`` 
inoltre
<Filtered a request with a potential harmful query string: ///remote/fgt_lang?lang=/../../../..//////////dev/>
e
`Logger: aiohttp.server
Source: /usr/local/lib/python3.9/site-packages/aiohttp/web_protocol.py:393
First occurred: 14:34:42 (2 occurrences)
Last logged: 14:34:54

Unhandled exception
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/asyncio/base_events.py", line 1183, in _sendfile_fallback
    read = await self.run_in_executor(None, file.readinto, view)
asyncio.exceptions.CancelledError

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/aiohttp/web_protocol.py", line 485, in start
    resp, reset = await task
  File "/usr/local/lib/python3.9/site-packages/aiohttp/web_protocol.py", line 440, in _handle_request
    reset = await self.finish_response(request, resp, start_time)
  File "/usr/local/lib/python3.9/site-packages/aiohttp/web_protocol.py", line 591, in finish_response
    await prepare_meth(request)
  File "/usr/local/lib/python3.9/site-packages/aiohttp/web_fileresponse.py", line 241, in prepare
    return await self._sendfile(request, fobj, offset, count)
  File "/usr/local/lib/python3.9/site-packages/aiohttp/web_fileresponse.py", line 96, in _sendfile
    await loop.sendfile(transport, fobj, offset, count)
  File "/usr/local/lib/python3.9/asyncio/base_events.py", line 1162, in sendfile
    return await self._sendfile_fallback(transport, file,
  File "/usr/local/lib/python3.9/asyncio/base_events.py", line 1192, in _sendfile_fallback
    await proto.restore()
  File "/usr/local/lib/python3.9/asyncio/base_events.py", line 263, in restore
    self._transport.resume_reading()
  File "/usr/local/lib/python3.9/asyncio/sslproto.py", line 343, in resume_reading
    self._ssl_protocol._transport.resume_reading()
AttributeError: 'NoneType' object has no attribute 'resume_reading'`

Scusami ancora per la lungaggine ma stò veramente brancollando nel buio più assoluto.
grazie

**Comment by fraygon** [comment]
Nella mio disperato tentativo di capirci qualcosa ho controllato le porte aperte del mio indirizzo ip e ho scoperto che la 443 è aperta ma la 8123 non è aperta. Vi chiedo se è normale o se anche quest'ultima deve essere vista aperta.

**Reply by snakuzzo** [reply]
Per funzionare tutto correttamente da app da internet devi:

1. Assicurarti di avere un certificato valido rilasciato da una CA
2. Assicurarti che il tuo FQDN sia risolto correttamente (associazione FQDN/indirizzp IP pubblico corretta)
3. Abilitare il portforwarding dalla porta esterna che ti pare verso IP raspberry su porta di ascolto di HA (default 8123)

Es. se punti ad `https://tuodominio.duckdns.org:9000` devi aprire una regola di portforwarding `9000 -> IP LAN di HA e porta 8123`

**Reply by fraygon** [reply]
ciao. grazie del tuo tempo.
Allora ricapitolando: 1. non capisco l'acronimo CA ma se intendi il certificato creato con duckdns è presente e valido.
2. anche qui sembra tutto a posto tanto è vero che riesco ad aprire HA con il pc e telefono tramite https ...duckdns.org
3. il portforwarding è regolarmente configurato con porta 443

Ribadisco che l'app funzionava alla perfezione fino a una settimana fa e non ho fatto nessuna modifica a parte aggiornamenti ufficiali di HA.

**Reply by snakuzzo** [reply]
CA = Certification Authority.
Quando apri tramite PC puntando il tuo FQDN ti dà un warning sul certificato oppure è tutto ok ?
Qual è l'errore che logga HA quando provi ad accedere da app?

**Reply by fraygon** [reply]
Quando apro con il pc è tutto regolare.
Ci sono due errori che compaiono e sono i seguenti
1: `Logger: aiohttp.server
Source: /usr/local/lib/python3.9/site-packages/aiohttp/web_protocol.py:393
First occurred: 12:03:12 (1 occurrences)
Last logged: 12:03:12

Unhandled exception
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/asyncio/base_events.py", line 1183, in _sendfile_fallback
    read = await self.run_in_executor(None, file.readinto, view)
asyncio.exceptions.CancelledError

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/aiohttp/web_protocol.py", line 485, in start
    resp, reset = await task
  File "/usr/local/lib/python3.9/site-packages/aiohttp/web_protocol.py", line 440, in _handle_request
    reset = await self.finish_response(request, resp, start_time)
  File "/usr/local/lib/python3.9/site-packages/aiohttp/web_protocol.py", line 591, in finish_response
    await prepare_meth(request)
  File "/usr/local/lib/python3.9/site-packages/aiohttp/web_fileresponse.py", line 241, in prepare
    return await self._sendfile(request, fobj, offset, count)
  File "/usr/local/lib/python3.9/site-packages/aiohttp/web_fileresponse.py", line 96, in _sendfile
    await loop.sendfile(transport, fobj, offset, count)
  File "/usr/local/lib/python3.9/asyncio/base_events.py", line 1162, in sendfile
    return await self._sendfile_fallback(transport, file,
  File "/usr/local/lib/python3.9/asyncio/base_events.py", line 1192, in _sendfile_fallback
    await proto.restore()
  File "/usr/local/lib/python3.9/asyncio/base_events.py", line 263, in restore
    self._transport.resume_reading()
  File "/usr/local/lib/python3.9/asyncio/sslproto.py", line 343, in resume_reading
    self._ssl_protocol._transport.resume_reading()
AttributeError: 'NoneType' object has no attribute 'resume_reading'
`
e il secondo
Logger: homeassistant.components.http.ban
Source: components/http/ban.py:124
Integration: HTTP (documentation, issues)
First occurred: 12:04:00 (1 occurrences)
Last logged: 12:04:00

`Logger: homeassistant.components.http.ban
Source: components/http/ban.py:124
Integration: HTTP (documentation, issues)
First occurred: 12:04:00 (1 occurrences)
Last logged: 12:04:00

Login attempt or request with invalid authentication from myfastgate.lan (192.168.1.254). (Mozilla/5.0 (Linux; Android 7.0; HUAWEI VNS-L31 Build/HUAWEIVNS-L31; wv) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/91.0.4472.164 Mobile Safari/537.36)`

**Reply by snakuzzo** [reply]
Serve l'errore quando NON riesci ad accedere da app, non quando accedi da PC

**Reply by fraygon** [reply]
adesso!!! non accedo più da duckdns. Ho aperto il sito di duck con il mio account e vedo che il token è diverso ed è stato generato 13 secondi fa.
Non capisco proprio cosa stà succedendo.

**Reply by snakuzzo** [reply]
vabbè...sostituisci il token con quello che avevi, riaggiorna associazione FQDN -> IP e riprova

**Reply by fraygon** [reply]
ok. porta un pò di pazienza ancora...
con il nuovo token e il nuovo FQDN ho interrogato il registro di duckdns e mi da tutto ok. Ho cambiato i valori su HA configuration e url esterno ma non accedo più, solo in locale apro HA. con il nuovo dominio si apre una pagina con il simbolo di HA e la scritta retry.
Mi hai scritto che devo associare FQDN con IP e vedo che duck mi ha assegnato un IP diverso dal precedente. 
Mi puoi dire dove eseguo questa associazione?

**Reply by snakuzzo** [reply]
Che sarebbe il registro di duckdns ?????
Quei valori url interno ed esterno su HA (e non sull'app) non servono a collegarti alla tua istanza. Servono alle integrazioni e custom component. Puoi pure lasciarli vuoti se non c'è necessità...non hanno niente a che vedere con la connessione da un client verso HA.
Se sul sito duckdns l'ip che vedi associato all'FQDN è uguale al tuo IP pubblico che ti ha assegnato il provider...è tutto ok.
L'associazione deve essere continua...e solitamente c'è un client che se ne occupa. Può essere il modem, un client installato da qualche parte, una chiamata REST ...dipende da quello che hai fatto tu...io non posso sapere cos'hai fatto.

**Reply by fraygon** [reply]
sono riuscito ad aprire HA tramite duckdns. Apro HA installato su raspberry che è collegato via LAN da computer  sia con URL interno sia con URL esterno.
Ma l'APP del telefono non ne vuole sapere. "Unable to connect to home Assistant"

**Reply by snakuzzo** [reply]
Devi dare evidenza di qualcosa...così non andiamo da nessuna parte :)
Quando provi a loggarti da app e ti dà "unable to connect" sul log di HA viene scritto qualcosa?
Devi guardare in tempo reale per capire se viene loggato proprio quando provi a loggarti su HA. Questo per capire se è un problema di HA oppure no.
Se logga qualcosa ... allora si prova a risolvere lato HA
Se non logga nulla...vuol dire che su HA nemmeno ci arrivi e il problema sarà a monte (proxy, router, etc etc)

**Reply by fraygon** [reply]
allora spento tutto, riacceso su pc e nessun log presente. Attivo la app e quasi immediatamente esce questo:
`Logger: homeassistant.components.http.ban
Source: components/http/ban.py:124
Integration: HTTP (documentation, issues)
First occurred: 19:39:48 (1 occurrences)
Last logged: 19:39:48

Login attempt or request with invalid authentication from myfastgate.lan (192.168.1.254). (Mozilla/5.0 (Linux; Android 7.0; HUAWEI VNS-L31 Build/HUAWEIVNS-L31; wv) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/91.0.4472.164 Mobile Safari/537.36)`

**Reply by snakuzzo** [reply]
Curiosità... hai un altro device da cui provare?

**Reply by fraygon** [reply]
si. un tablet. stesso identico problema.

**Reply by snakuzzo** [reply]
Sono device vecchi?
Poi perdonami...fai la prova da browser sia su tablet che su celle e vedi se è tutto ok (certificato compreso)

**Reply by fraygon** [reply]
telefono 5anni, tablet 3 anni.
sono sconcertato.... aprendo chrome non si apre da nessuna parte. Sul tablet ho installato anche firefox focus e con quest'ultimo si apre. Te lo spieghi?

**Reply by snakuzzo** [reply]
Che modelli sono di preciso?

**Reply by fraygon** [reply]
huawei p9 lite telefono e asus zenpad il tablet

**Reply by maxcanna** [reply]
Puoi provare ad inserire il tuo dominio qui https://www.ssllabs.com/ssltest/ e avviare il test? Poi incolla il risultato (fai uno screen). Ricordati di cancellare il tuo dominio se no lo vedono tutti

**Reply by fraygon** [reply]
scusami ma ci sono un sacco di informazioni. Devo inviare tutto? saranno come minimo 10 screen

**Reply by maxcanna** [reply]
Devi controllare se gli handshake con android sono andati a buon fine

**Reply by fraygon** [reply]
sembra proprio di si. su Handshake Simulation non ci sono errori. Su # Not simulated clients (Protocol mismatch) solo android 2.3.7 presenta errore no SNI.
Il test mi ha dato un rating A.

**Reply by fraygon** [reply]
Vi ringrazio dell'assistenza ma ora stò crollando. Dopo un pomeriggio passato a tentare di capire qualcosa non ce la faccio più. SE avete altri suggerimenti non esitate a darmeli. Anche se ci capisco ben poco di reti, web, protocolli voglio provare ancora prima di reinstallare tutto da zero. 
buonanotte

**Reply by fraygon** [reply]
Ho installato la app su un android 10 e funziona regolarmente. Chiedo alla community se anche voi avete problemi con android 7. Mi sembra strano, non sono versioni proprio vecchie. Alla faccia della obsolescenza programmata!!.

**Reply by snakuzzo** [reply]
Lasciando perdere i discorsi sull' "obsolescenza programmata", qual è la versione dell'app che non va ?
In ogni caso potresti provare a chiedere info direttamente sul canale discord di HA dove ci sono anche gli sviluppatori.

**Reply by fraygon** [reply]
nel forum di HA ho trovato un'altro con il mio stesso problema, mi sono aggiunto e speriamo che venga risolto il problema. Cambiare devices per questo mi sembra troppo. 
La versione è quella pubblicata su play store: 2021.6.2-full

**Reply by KC-inDomus** [reply]
puoi sempre usare il browser su android 7, anziché l'app. e come tracker, qualcos'altro

**Reply by fraygon** [reply]
la comodità dell'APP è impagabile. Per il momento ho risolto installando una app chiamata WallPanel che è nata per visualizzare, tra le altre, anche Home assistant su vecchi tablet android. Funziona molto bene. 
Certo che c'è da dannarsi quando ti capita un problema come il mio e non capendo una "cippa" di funzionamento web, internet etcc. 
Fortuna che ci siete voi a spronarci.

**Reply by snakuzzo** [reply]
sì ma ti cambia poco rispetto ad usarla via browser. Ti perdi giusto le notifiche native (sostituibili da quelle Telegram ad esempio) ed il device tracker GPS (sostituibile con app terze parti tipo life360 che va pure meglio del tracker dell'app)

**Reply by fraygon** [reply]
Tanto per chiudere questo post vi informo degli ultimi sviluppi, nel caso qualcun'altro si trovi i miei stessi problemi.
In pratica sembra che ai miei dispositivi manchi l'aggiornamento dell'app "android system webview". Google non mi fa aggiornare questa sui dispositivi android 7. Le farie faq di google parlano di problemi a android 8 (non tutti i dispositivi). Nessuno sa come risolvere.
A mio modesto parere ribadisco che questa E' obsolescenza programmata che si fa sia sui prodotti con l'hardware e, più subdolamente, via software proprietari.

**Reply by NdR91** [reply]
Per come la vedo io, parlare di obsolescenza programmata su un software risalente al 2016 è totalmente fuori luogo. 

Hai un software obsoleto, questo è quanto.

**Reply by snakuzzo** [reply]
Concordo con @ndr91 ... non facciamo i complottisti 😅
